flink-runtime/src/main/java/org/apache/flink/runtime/operators/coordination/SubtaskGatewayImpl.java

```diff
@@ -86,6 +91,24 @@ class SubtaskGatewayImpl implements OperatorCoordinator.SubtaskGateway {
 
     @Override
     public CompletableFuture<Acknowledge> sendEvent(OperatorEvent evt) {
+        return sendEventWithCallBackOnCompletion(
+                evt,
+                (success, failure) -> {
+                    if (failure != null && subtaskAccess.isStillRunning()) {
```

Comment:
Should we use `failure instanceof TaskNotRunningException` to check whether the task is running?